src/components/common/Modal.vue

```diff
@@ -0,0 +1,182 @@
+              v-if="errors.height"
+            />
+          </text-input>
+          <radio-group
+            :options="heightRadioOptions"
+            classes="ml-8 w-36"
+            group-label="Jednostka"
+            v-model:selected="heightUnit"
+          />
+        </div>
+        <div class="flex justify-center items-end mx-12 mb-8">
+          <text-input
+            classes="relative w-24"
+            type="number"
+            label="Waga"
+            min="0"
+            v-model:value.number="weight"
+          >
+            <Error
+              :message="errors.weight"
+              classes="absolute w-max -bottom-6"
+              v-if="errors.weight"
+            />
+          </text-input>
+          <radio-group
+            :options="weightRadioOptions"
+            classes="relative ml-8 w-36"
+            group-label="Jednostka"
+            v-model:selected="weightUnit"
+          />
+        </div>
+        <Button class="border border-black mx-12 my-6" label="Zapisz" />
+      </form>
+    </div>
+  </div>
+</template>
+
+<script>
+import Avatar from "@/components/Avatar";
+import Button from "@/components/common/Button";
+import Error from "@/components/common/Error";
+import RadioGroup from "@/components/common/RadioGroup";
+import TextInput from "@/components/common/TextInput";
+import { ref } from "vue";
+import { useStore } from "vuex";
+
+export default {
+  name: "Modal",
+  components: {
+    Avatar,
+    Button,
+    Error,
+    RadioGroup,
+    TextInput,
+  },
+  setup() {
+    const store = useStore();
+    const name = ref(store.getters.getName);
+    const lastname = ref(store.getters.getLastname);
+    const height = ref(store.getters.getHeight);
+    const weight = ref(store.getters.getWeight);
+    const heightUnit = ref(store.getters.getHeightUnit);
+    const weightUnit = ref(store.getters.getWeightUnit);
+    const heightRadioOptions = [
+      {
+        value: "m",
+        label: "m",
+      },
+      {
+        value: "cm",
+        label: "cm",
+      },
+    ];
+    const weightRadioOptions = [
+      {
+        value: "kg",
+        label: "kg",
+      },
+      {
+        value: "lbs",
+        label: "lbs",
+      },
+    ];
+    const errors = ref({});
+
+    const checkForm = (e) => {
+      errors.value = {};
+      e.preventDefault();
+      if (!/^\D{2,}$/.test(name.value) || !name.value) {
+        errors.value.name =
+          "Imię musi posiadać co najmniej 2 litery i żadnych cyfr";
+      }
+      if (!/^\D{2,}$/.test(lastname.value) || !lastname.value) {
+        errors.value.lastname =
+          "Nazwisko musi posiadać co najmniej 2 litery i żadnych cyfr";
+      }
+      if (!height.value) {
+        errors.value.height = "Wzrost jest wymagany";
+      }
+      if (!weight.value) {
+        errors.value.weight = "Waga jest wymagana";
+      }
+      if (Object.keys(errors.value).length === 0) setUser();
+    };
+    const hideModal = () => store.commit("HIDE_MODAL");
```

Comment:
lepiej niech to emmituje event i rodzic go obsługuje - tzn w parencie jakiś v-if czy ma być modal open czy nie, i zmieniasz tę zmienną za pomocą eventu.
stora lepiej do tego nie używać, bo możesz mieć 15 modali na stronie i do kazdego byś musiał wtedy robić osobny var w storze